sdk/cognitiveservices/Azure.ResourceManager.CognitiveServices/src/autorest.md

```diff
@@ -137,6 +136,7 @@ directive:
       $.CheckDomainAvailabilityParameter.properties.type['x-ms-format'] = 'resource-type';
       $.CheckSkuAvailabilityParameter.properties.type['x-ms-format'] =  'resource-type';
       $.PrivateEndpointConnection.properties.properties['x-ms-client-flatten'] = true;
+      $.ModelSku.properties.rateLimits['readOnly'] = true;
```

Comment:
> I understand it is a breaking change from C# side.
Why non-breaking change in swagger generate breaking change code in SDK. Is it something can be fixed from C# side.

@yangyuan This change in swagger caused the breaking change in .NET SDK, do you want to change it back? If so, should it be changed in the spec repo instead?
If the business does require the change here, you can do what I suggested before. Rename the new property and add the old one back to custom code and make it EB.Never and obsolete.

Reply:
I believe this is a code-gen's fault. A non-breaking change in swagger causing a breaking change in SDK.

I did some tests.
Our swagger didn't specific mark this field as readOnly, but all the objects using this field are either readOnly, or only as a get result (which is also read only).
Isn't read only implied in here? looks like code gen can recognize the first case but not the second.

I'm sure more and more teams will hit this issue in future.
Since swagger is already published, we cannot add readOnly anymore, as its breaking change.

Reply:
This breaking change was indeed caused by the generator.
I will investigate if we can afford the impact of changing the generator's behavior.